hack/ci/windows.ps1

```diff
@@ -1012,9 +1012,6 @@ Catch [Exception] {
     Throw $_
 }
 Finally {
-    # Preserve the LastExitCode of the tests
-    $tmpLastExitCode = $LastExitCode
-
     $ErrorActionPreference="SilentlyContinue"
```

Comment:
I guess, the `Catch [Exception]` already should handle any error that occurred so far, and the `SilentlyContinue` won't mask those errors, correct?

Reply:
Inside the `Try` everything is set up to Throw as a fail-out case, and then the `Catch` logs and it either turns it into an error code with `exit` for CI, or `Throw`s onwards for interactive use. As far as I could tell by testing (I couldn't find any good docs on this) `Finally` can't interfere with the `exit` result unless it also calls `exit`.

(I suspect `exit` is internally a `Throw` with a specific type caught at the top-level, based on the above, somewhat like Python's `sys.exit()`.)

I assume the `SilentlyContinue` here is because we don't want the `Finally` block to change the test outcome by throwing when we otherwise passed (most of the things in this block also explicitly `SilentlyContinue` anyway, like the log-collection) since for CI, most of the things we're doing here are not interesting (cleanups) and for interactive use, you wouldn't want a cleanup failure to overwrite a test failure result.

It might make sense to use `Continue` instead of `SilentlyContinue` so that failures in this block log their error messages, perhaps.